### PR TITLE
Use six to support Python 2 and 3 without using 2to3

### PR DESCRIPTION
**Pull request by davehunt**

### CODE REVIEW

File: blessings/__init__.py [hunk 1]
```diff
@@ -420,7 +422,7 @@ def derivative_colors(colors):
                       'shadow', 'standout', 'subscript', 'superscript']))
 
 
-class ParametrizingString(unicode):
+class ParametrizingString(six.text_type):
```
Comment (erikrose) [inline blessings/__init__.py]: My only comment at a glance is to import these symbols more deeply to match the rest of them, as in `from six import text_type, ...`.

Comment (davehunt) [reply blessings/__init__.py]: I'm entirely not sure why, but doing this fails for me when running the tests, with the following traceback:

```
GLOB sdist-make: /Users/dhunt/workspace/blessings/setup.py
py27 inst-nodeps: /Users/dhunt/workspace/blessings/.tox/dist/blessings-1.6.1.zip
py27 installed: blessings==1.6.1,nose==1.3.7,six==1.11.0
py27 runtests: PYTHONHASHSEED='3106176075'
py27 runtests: commands[0] | nosetests blessings
...............S...E...
======================================================================
ERROR: Make sure you get nice hints if you misspell a formatting wrapper.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/dhunt/workspace/blessings/.tox/py27/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/dhunt/workspace/blessings/blessings/tests.py", line 232, in test_nice_formatting_errors
    t.bold_misspelled('hey')
  File "/Users/dhunt/workspace/blessings/blessings/__init__.py", line 466, in __call__
    if len(args) == 1 and isinstance(args[0], string_types):
NameError: global name 'string_types' is not defined

----------------------------------------------------------------------
Ran 23 tests in 0.006s

FAILED (SKIP=1, errors=1)
ERROR: InvocationError for command '/Users/dhunt/workspace/blessings/.tox/py27/bin/nosetests blessings' (exited with code 1)
```

It's the same for all environments.

File: blessings/tests.py [hunk 2]
```diff
@@ -10,13 +10,14 @@
 
 """
 from __future__ import with_statement  # Make 2.5-compatible
+from __future__ import absolute_import
```
Comment (erikrose) [inline blessings/tests.py]: Is this (and the one in the other file) just for hygiene? I don't see that it changes any behavior in the context of this project.

Comment (erikrose) [reply blessings/tests.py]: Also, it doesn't do anything in Python 2.7 and later, which is all we support now. :-) https://www.python.org/dev/peps/pep-0328/

File: blessings/__init__.py (hunk 1 above)
Comment (erikrose) [reply blessings/__init__.py]: Hmm, I tried it locally, and it worked for me. Maybe if you push it, it'll work on Travis.

**Comment by davehunt** [comment]
Sorry for the delay, I had this work thing last week that was rather distracting. 😉 I only saw you once, from a distance, otherwise I'd have said hello 👋 and thanked you for your recent help!

**Comment by erikrose** [comment]
Thanks, Dave! I forget what your original motivation was. Do you want me to release a new version?

**Comment by davehunt** [comment]
> Do you want me to release a new version?

Yes please, once it's been released I'll be updating the vendored copy in mozilla-central and using it to start running some tests against Python 3.

**Comment by erikrose** [comment]
Done! https://pypi.org/project/blessings/#files

**Pull request merged**